dotnet/src/Experimental/Process.LocalRuntime/LocalStep.cs

```diff
@@ -254,13 +254,17 @@ protected virtual async ValueTask InitializeStepAsync()
                 throw new KernelException(errorMessage);
             }
 
-            stateObject = (KernelProcessStepState?)Activator.CreateInstance(stateType, this.Name, this.Id);
+            var userState = stateType.GetProperty(nameof(KernelProcessStepState<object>.State))?.GetValue(stateObject);
+            if (userState is null)
+            {
+                stateType.GetProperty(nameof(KernelProcessStepState<object>.State))?.SetValue(stateObject, Activator.CreateInstance(userStateType));
```

Comment:
with this line then any class using KernelProcessStep `this._state ??= state.State ?? new();` -> **new()** is not necessary right only `this._state ??= state.State` ?

Just wondering if at some point the override `ValueTask ActivateAsync` function is not necessary to get overridden by all steps with a type

Reply:
> with this line then any class using KernelProcessStep `this._state ??= state.State ?? new();` -> **new()** is not necessary right only `this._state ??= state.State` ?

Yes, totally right, thanks!